### PR TITLE
Refactor MyTgClientDebug

### PR DESCRIPTION
**Pull request by wkejqb**
- rename to StatusWatcherScenario
 - remove second key requirement

### CODE REVIEW

File: examples/client.php [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+use TelegramOSINT\Scenario\StatusWatcherScenario;
 
 require_once __DIR__.'/../vendor/autoload.php';
 
 // here we get contact list and get contact online status
 // avatars are saved to current directory
 
 /* @noinspection PhpUnhandledExceptionInspection */
-(new MyTgClientDebug())->startActions();
+(new StatusWatcherScenario())->startActions();
```
Comment (anton-postuf) [inline examples/client.php]: Rename client.php to watcher.php

File: src/Scenario/AbstractGroupScenario.php [hunk 2]
```diff
@@ -27,7 +27,7 @@ public function setDeepLink(string $deepLink): void
 
     public function __construct(?Proxy $proxy = null, ?ClientGeneratorInterface $generator = null)
     {
-        parent::__construct($proxy, $generator);
+        parent::__construct($generator, $generator, $proxy);
```
Comment (anton-postuf) [inline src/Scenario/AbstractGroupScenario.php]: Two generators passing, is it ok?

File: src/Scenario/ClientGenerator.php [hunk 3]
```diff
@@ -22,13 +22,8 @@ public function getStatusWatcherClient(StatusWatcherCallbacks $callbacks)
         return new StatusWatcherClient($callbacks);
     }
 
-    public function getAuthKeyInfo(): string
+    public function getAuthKey(): string
     {
         return trim(file_get_contents('./first.authkey'));
```
Comment (anton-postuf) [inline src/Scenario/ClientGenerator.php]: It is better to iliminate default values - let user pass it madatory

File: src/Scenario/StatusWatcherScenario.php [hunk 4]
```diff
@@ -21,9 +21,9 @@
  *
  * Uses two telegram, connections (infoClient, monitoringClient).
  *
- * Requires files: `first.authkey`, `second.authkey`.
+ * Requires files: `first.authkey`.
  */
-class MyTgClientDebug implements StatusWatcherCallbacks, ClientDebugLogger, ScenarioInterface
+class StatusWatcherScenario implements StatusWatcherCallbacks, ClientDebugLogger, ScenarioInterface
```
Comment (anton-postuf) [inline src/Scenario/StatusWatcherScenario.php]: This client could be redundant, possibly could be removed

File: src/Scenario/AbstractGroupScenario.php (hunk 2 above)
Comment (tombouctou) [reply src/Scenario/AbstractGroupScenario.php]: ok, the second one is not really used in this case

File: src/Scenario/StatusWatcherScenario.php (hunk 4 above)
Comment (tombouctou) [reply src/Scenario/StatusWatcherScenario.php]: let's leave it for now, we have scenario for status watching

File: examples/client.php (hunk 1 above)
Comment (wkejqb) [reply examples/client.php]: done

File: src/Scenario/StatusWatcherScenario.php (hunk 4 above)
Comment (anton-postuf) [reply src/Scenario/StatusWatcherScenario.php]: Lets get rid of info client then - it is redundant

File: src/Scenario/ClientGenerator.php (hunk 3 above)
Comment (wkejqb) [reply src/Scenario/ClientGenerator.php]: passed via env var now, required.

File: src/Scenario/StatusWatcherScenario.php (hunk 4 above)
Comment (wkejqb) [reply src/Scenario/StatusWatcherScenario.php]: done

File: src/Scenario/ClientGenerator.php [hunk 5]
```diff
@@ -9,9 +9,18 @@
...
 use TelegramOSINT\Client\StatusWatcherClient\StatusWatcherCallbacks;
 use TelegramOSINT\Client\StatusWatcherClient\StatusWatcherClient;
+use TelegramOSINT\Exception\TGException;
 
 class ClientGenerator implements ClientGeneratorInterface
 {
+    /** @var string */
+    private $envName;
+
+    public function __construct(string $envName = 'AUTHKEY')
```
Comment (anton-postuf) [inline src/Scenario/ClientGenerator.php]: Would be better make it mandatory, not optional

File: src/Scenario/ClientGenerator.php [hunk 6]
```diff
@@ -22,13 +31,18 @@ public function getStatusWatcherClient(StatusWatcherCallbacks $callbacks)
...
+     * @return string
+     */
+    public function getAuthKey(): string
     {
-        return trim(file_get_contents('./first.authkey'));
-    }
+        $envPath = getenv($this->envName);
+        if (!$envPath || !file_exists($envPath)) {
+            throw new TGException(0, "Please set {$this->envName} env var to valid filename");
+        }
```
Comment (anton-postuf) [inline src/Scenario/ClientGenerator.php]: Better to keep in ENV-var authkey itself, no file pathes

File: src/Scenario/InfoClientScenario.php [hunk 7]
```diff
@@ -13,20 +14,39 @@ abstract class InfoClientScenario implements ScenarioInterface
...
+     * @throws TGException
+     */
+    protected function infoLogin(): void
+    {
+        $authKey = AuthKeyCreator::createFromString($this->authKey);
+        if (!$this->infoClient->isLoggedIn()) {
+            $this->infoClient->login($authKey);
+        }
+    }
+
```
Comment (anton-postuf) [inline src/Scenario/InfoClientScenario.php]: Get rid of this method - "login" only should exists

File: src/Scenario/ClientGenerator.php (hunk 6 above)
Comment (wkejqb) [reply src/Scenario/ClientGenerator.php]: Let's leave env var for now, to be fixed in future refactoring.

File: src/Scenario/InfoClientScenario.php (hunk 7 above)
Comment (wkejqb) [reply src/Scenario/InfoClientScenario.php]: Fixed, renamed to `login`

File: src/Scenario/ClientGenerator.php (hunk 5 above)
Comment (wkejqb) [reply src/Scenario/ClientGenerator.php]: Ok, made mandatory, env var is now selected at 3rd/4th level.

File: src/Scenario/InfoClientScenario.php (hunk 7 above)
Comment (wkejqb) [reply src/Scenario/InfoClientScenario.php]: Renamed to `login`.

File: src/Scenario/StatusWatcherScenario.php [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+        $this->client->login($authKey, $this->proxy);
+
+        /* add via phone numbers */
+        $monitoringPhones = [
+            '541123683798',
+            '60192003400',
+            '77779479694',
+            '393475723072',
+            '989196190933',
+        ];
```
Comment (anton-postuf) [inline src/Scenario/StatusWatcherScenario.php]: These numbers shoul not be hardcoded

File: src/Scenario/UserContactsScenario.php [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    /**
+     * @throws TGException
+     */
+    protected function getContactsInfo(): void
+    {
+        $phones = [
+            '+79153801634',
+        ];
```
Comment (anton-postuf) [inline src/Scenario/UserContactsScenario.php]: These numbers shoul not be hardcoded

File: src/Scenario/StatusWatcherScenario.php (hunk 8 above)
Comment (tombouctou) [reply src/Scenario/StatusWatcherScenario.php]: right, removed those, now example awaits number list in argv.

File: src/Scenario/UserContactsScenario.php (hunk 9 above)
Comment (tombouctou) [reply src/Scenario/UserContactsScenario.php]: right, removed those, now example awaits number list in argv.

**Pull request merged**